assets/src/edit-story/components/canvas/useCanvasSelectionCopyPaste.js

```diff
@@ -108,41 +111,108 @@ function useCanvasSelectionCopyPaste(container) {
     [deleteSelectedElements, selectedElements]
   );
 
+  const elementPasteHandler = useCallback(
+    (html) => {
+      const template = document.createElement('template');
+      template.innerHTML = html;
+      let foundElements = false;
+      for (let n = template.content.firstChild; n; n = n.nextSibling) {
+        if (n.nodeType !== /* COMMENT */ 8) {
+          continue;
+        }
+        const payload = JSON.parse(
+          n.nodeValue.replace(new RegExp(DOUBLE_DASH_ESCAPE, 'g'), '--')
+        );
+        if (payload.sentinel !== 'story-elements') {
+          continue;
+        }
+        foundElements = true;
+        payload.items.forEach(({ x, y, basedOn, ...rest }) => {
+          currentPage.elements.forEach((element) => {
+            if (element.id === basedOn || element.basedOn === basedOn) {
+              x = Math.max(x, element.x + 60);
+              y = Math.max(y, element.y + 60);
+            }
+          });
+          const element = {
+            ...rest,
+            basedOn,
+            id: uuidv4(),
+            x,
+            y,
+          };
+          addElement({ element });
+        });
+      }
+      return foundElements;
+    },
+    [addElement, currentPage]
+  );
+
+  // @todo Split up this part to make the file smaller/more manageable.
+  const allowedContentNodes = ['strong', 'em', 'u'];
+
+  const processNodeContent = useCallback(
+    (node) => {
+      const tag = node.parentNode?.tagName?.toLowerCase();
+      const stripTags = !allowedContentNodes.includes(tag);
+      if (stripTags) {
+        return node.textContent.trim();
+      }
+      return `<${tag}>${node.textContent.trim()}<${tag}/>`;
```

Comment:
The content in between `<>...</>` needs to be escaped here: e.g. "Use &lt;u&ht; for underline" would turn into "Use <u> for underline" in the `textContent` and that's not something we can use for HTML. We need to transform it back to `&lt;` and such.